subprojects/migration/src/main/java/org/mockito/migration/MockitoV1CompatibilityMatcher.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2017 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.migration;
+
+import java.lang.reflect.Method;
+import org.mockito.ArgumentMatcher;
+import org.mockito.Incubating;
+
+/**
+ * Base class for classes that implement {@link ArgumentMatcher}.
+ * <p>
+ * This was provided to simplify migration from pre 2.x implementations; it resolves the conflict
+ * caused by the signature of the {@code matches(...)} method changing from {@code matches(Object)}
+ * in pre 2.x to {@code matches(T)} by adding a new {@link #matchesSafely(T)} method which
+ * developers implement instead. This checks the type of the
+ * argument beforehand and returns false if it is not of the correct type to avoid a
+ * {@link ClassCastException} being thrown.
+ * <p>
+ * It does not provide a default implementation of {@link #toString()} as if no {@link #toString()}
+ * method is provided then it defaults to the pre 2.x behavior of creating a decamelized form of the
+ * class name.
+ *
+ * @deprecated this was only intended to ease migration from 1.x to 2.x please implement
+ * {@link ArgumentMatcher} directly.
```

Comment:
Let's add @since tag here. If you think that we're missing @since tag somewhere else, let's add it, too :)